pkg/roachpb/api.go

```diff
@@ -771,13 +771,32 @@ func NewReverseScan(key, endKey Key) Request {
 	}
 }
 
-func (*GetRequest) flags() int                { return isRead | isTxn }
-func (*PutRequest) flags() int                { return isWrite | isTxn | isTxnWrite }
-func (*ConditionalPutRequest) flags() int     { return isRead | isWrite | isTxn | isTxnWrite }
-func (*InitPutRequest) flags() int            { return isRead | isWrite | isTxn | isTxnWrite }
-func (*IncrementRequest) flags() int          { return isRead | isWrite | isTxn | isTxnWrite }
-func (*DeleteRequest) flags() int             { return isWrite | isTxn | isTxnWrite }
-func (*DeleteRangeRequest) flags() int        { return isWrite | isTxn | isTxnWrite | isRange }
+func (*GetRequest) flags() int            { return isRead | isTxn }
+func (*PutRequest) flags() int            { return isWrite | isTxn | isTxnWrite }
+func (*ConditionalPutRequest) flags() int { return isRead | isWrite | isTxn | isTxnWrite }
+func (*InitPutRequest) flags() int        { return isRead | isWrite | isTxn | isTxnWrite }
+func (*IncrementRequest) flags() int      { return isRead | isWrite | isTxn | isTxnWrite }
+func (*DeleteRequest) flags() int         { return isWrite | isTxn | isTxnWrite }
+func (drr *DeleteRangeRequest) flags() int {
+	// DeleteRangeRequest has different properties if the "inline" flag is set.
+	// This flag indicates that the request is deleting inline MVCC values,
+	// which cannot be deleted transactionally - inline DeleteRange will thus
+	// fail if executed as part of a transaction. This alternate flag set
+	// is needed to prevent the command from being automatically wrapped into a
+	// transaction by TxnCoordSender, which can occur if the command spans
+	// multiple ranges.
+	//
+	// TODO(mrtracy): The behavior of DeleteRangeRequest with "inline" set has
+	// likely diverged enough that it should be promoted into its own command.
+	// However, it is complicated to plumb a new command through the system,
```

Comment:
@petermattis I've expanded the comment on that field of DeleteRangeRequest.

@vivekmenezes the code that ends up triggering the transaction depends on the properties of the _batch_ being executed, not the specific command.  The batch's flags are computed from its component commands, and can result in a non-transaction command being upgraded to transactional.  So, not only do we want inline DeleteRange to be non-transactional, but we also have to make sure that it isn't grouped into a batch with a transactional command - that means we also have to set the `isAlone` flag on the command.  The `flags()` command allows us to do both of those things.